event-sourcing/event-sourcing-core/src/main/scala/org/apache/james/eventsourcing/CommandDispatcher.scala

```diff
@@ -45,9 +49,19 @@ object CommandDispatcher {
 class CommandDispatcher @Inject()(eventBus: EventBus, handlers: Set[CommandHandler[_ <: Command]]) {
   Preconditions.checkArgument(hasOnlyOneHandlerByCommand(handlers), CommandDispatcher.ONLY_ONE_HANDLER_PRECONDITION)
 
-  def dispatch(c: Command): Unit = {
-    trySeveralTimes(() => tryDispatch(c))
-      .getOrElse(() => throw CommandDispatcher.TooManyRetries(c, CommandDispatcher.MAX_RETRY))
+  def dispatch(c: Command): Publisher[Unit] = {
```

Comment:
Sometime I see `Publisher[Unit]`, sometime `Publisher[Void]`. I think we should use a unified type for scala code